### PR TITLE
Made schema inference work with `s3Cluster` table function

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added a support for automatic schema inference to `s3Cluster` table function. Synced the signatures of  `s3 ` and `s3Cluster`

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/TableFunctions/TableFunctionS3Cluster.cpp [hunk 1]
```diff
@@ -45,102 +46,104 @@ void TableFunctionS3Cluster::parseArguments(const ASTPtr & ast_function, Context
...
+    std::copy(args.begin() + 1, args.end(), std::back_inserter(clipped_args));
 
-    if (args_to_idx.contains("secret_access_key"))
-        secret_access_key = args[args_to_idx["secret_access_key"]]->as<ASTLiteral &>().value.safeGet<String>();
+    /// StorageS3ClusterConfiguration inherints from StorageS3Configuration, so it is safe to upcast it.
+    TableFunctionS3::parseArgumentsImpl(message, clipped_args, context, static_cast<StorageS3Configuration & >(configuration));
 }
 
 
 ColumnsDescription TableFunctionS3Cluster::getActualTableStructure(ContextPtr context) const
```
Comment (Avogar) [inline src/TableFunctions/TableFunctionS3Cluster.cpp:83]: We should also add schema inference here, look at the `TableFunctionS3::getActualTableStructure`. It's needed to be able to run queries like 'desc s3Cluster(...)' that will call `getActualTableStructure` without storage creation.

**Comment by qoega** [comment]
And do you want to add tests?

**Comment by nikitamikhaylov** [comment]
> And do you want to add tests?

Sure.

**Comment by Avogar** [comment]
Let's add tests with `desc s3Cluster(...)`

**Comment by alexey-milovidov** [comment]
What about hdfsCluster, hiveCluster?

**Comment by Avogar** [comment]
> What about hdfsCluster, hiveCluster?

They don't support schema inference yet, but we know about them and will look at it.

**Comment by nikitamikhaylov** [comment]
@Avogar I know, because I didn't update the reference file

**Comment by nikitamikhaylov** [comment]
BTW: hiveCluster is not merged yet

**Review by Avogar — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
@Mergifyio update

**Comment by Avogar** [comment]
`Stateless tests (debug, actions) [2/3] — Timeout, fail: 1, passed: 923, skipped: 7` - it's related to changes.
```sql
SELECT * FROM s3(decodeURLComponent(NULL), [NULL])
```
leads to logical error:
```
Logical error: 'Bad cast from type DB::ASTFunction to DB::ASTLiteral'
```

**Comment by nikitamikhaylov** [comment]
@Mergifyio update

**Pull request merged**